streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -408,9 +417,74 @@ public void shouldProcessInOrder() {
         assertEquals(3, source2.numReceived);
     }
 
+    @Test
+    public void shouldProcessRecordsAfterPrepareCommitWhenEosDisabled() {
+        task = createStatelessTask(createConfig(), StreamsConfig.METRICS_LATEST);
```

Comment:
nit: we may extract the common test logic to be reused by all 3 process record tests.

Reply:
Thinking about it, the common test logic is so small, it seems to make the test hard to read if we extract it?